native/src/fairseq2n/data/text/string_splitter.cc

```diff
@@ -17,17 +17,26 @@ using namespace fairseq2n::detail;
 
 namespace fairseq2n {
 
+static std::vector<std::size_t>
+wrap_indices(std::variant<std::size_t, std::vector<std::size_t>> &&indices) {
+    if (std::holds_alternative<std::size_t>(indices)) {
+        return {std::get<std::size_t>(std::move(indices))};
+    } else {
+        return std::get<std::vector<std::size_t>>(std::move(indices));
+    }
+}
+
 string_splitter::string_splitter(
     char separator,
     std::vector<std::string> names,
-    std::vector<std::size_t> indices,
+    std::variant<std::size_t, std::vector<std::size_t>> indices,
     bool exclude)
-  : separator_{separator}, names_(std::move(names)), indices_{std::move(indices)}, exclude_{exclude}
+  : separator_{separator}, names_(std::move(names)), indices_{wrap_indices(std::move(indices))}, exclude_{exclude}, single_column_{std::holds_alternative<std::size_t>(indices)}
```

Comment:
Do you mind also limiting your lines to 100 chars as long as it does not reduce readability?

Reply:
Done.